internal/provider/user_resource.go

```diff
@@ -274,6 +275,23 @@ func (r *UserResource) Read(ctx context.Context, req resource.ReadRequest, resp
 	data.LoginType = types.StringValue(string(user.LoginType))
 	data.Suspended = types.BoolValue(user.Status == codersdk.UserStatusSuspended)
 
+	// Also query by username to check for deletion or username reassignment
+	userByName, err := client.User(ctx, data.Username.ValueString())
+	if err != nil {
+		if isNotFound(err) {
+			resp.Diagnostics.AddWarning("Client Warning", fmt.Sprintf("User with ID %q not found. Marking as deleted.", data.ID.ValueString()))
```

Comment:
Shouldn't this read "user with username not found"? These errors should probably be more descriptive and say that the terraform-managed user was deleted outside of terraform